src/runtime_src/core/pcie/tools/xbutil/xbutil.cpp

```diff
@@ -918,6 +918,17 @@ void testCaseProgressReporter(bool *quit)
     }
 }
 
+void set_shell_path_env(const std::string& var_name, const std::string& trailing_path, int overwrite)
+{
+    std::string xrt_path(getenv("XILINX_XRT"));
```

Comment:
This is not safe.  getenv can return nullptr.

I strongly recommend this simple helper function:
```
const char* value_or_empty(const char* value) { return value ? value : "" ; }
std::string get_env(const char* env) { return value_or_empty(std::getenv(env)); }
```
now you can get env without explicitly checking the return value of std::getenv.
`std::string xrt_path(get_env("XILINX_XRT"));`

Also, xrt_path doesn't change per call, so make it static within function so as to compute just once.